packages/timelock-manager/contracts/TimelockManager.sol

```diff
@@ -11,24 +11,27 @@ import "./interfaces/ITimelockManager.sol";
 /// @title Contract that timelocks API3 tokens sent to it until the vesting
 /// period is over or the staking pool is operational
 /// @notice The owner of TimelockManager (i.e., API3 DAO) can send tokens to
-/// TimelockManager to be timelocked until releaseTime. After releaseTime, the
-/// respective owner can withdraw the tokens.
+/// TimelockManager to be timelocked until releaseStart. After releaseStart, the
+/// respective owner can withdraw the tokens, either immedietly or on a cliff.
 /// Alternatively, if the owner of this contract sets api3Pool, the token
 /// owners can transfer their tokens from TimelockManager to api3Pool before
-/// releaseTime. These tokens will be not be withdrawable from api3Pool until
+/// releaseStart. These tokens will be not be withdrawable from api3Pool until
 /// their respective releaseTimes.
 /// API3 DAO can also reverse timelocks (i.e., annul them) and send the tokens
 /// to a destination of its choice. Note that timelocks can be specified not to
 /// be reversible.
 contract TimelockManager is Ownable, ITimelockManager {
     using SafeMath for uint256;
 
-    struct Timelock {
+    struct Timelock { 
         address owner;
-        uint256 amount;
-        uint256 releaseTime;
+        uint256 totalAmount;
+        uint256 releasedAmount;
+        uint256 releaseStart;
+        uint256 releaseEnd;  
+        uint256 cliffTime;  
```

Comment:
The cliff can be implemented only using `releaseStart` and `releaseEnd`. For example if we want a 6 month cliff and a 18 month vesting schedule:
```
releaseStart = now + 6 months
releaseEnd = now + 6 months + 18 months
``` 
There's no need for a separate `cliffTime`.